zoo-project/zoo-kernel/service_internal_python.c

```diff
@@ -942,20 +942,61 @@ PythonUpdateStatus(PyObject* self, PyObject* args)
  * @return a new Python string containing the translated value
  * @see _ss
  */
-#include "frameobject.h"
+#include <Python.h>
+
+// This function prints a debug message along with the module, function, and line number
+// where it was called from. Only string arguments are supported.
 PyObject*
 PythonPrintDebugMessage(PyObject* self, PyObject* args)
 {
-  char *str;
-  if (!PyArg_ParseTuple(args, "s", &str)){
-    fprintf(stderr,"Incorrect arguments to debug, only strings are supported\n");
-  }
-  PyFrameObject* poFrame=PyEval_GetFrame();
-  char const* pccModule = _PyUnicode_AsString(poFrame->f_code->co_filename);
-  char const* pccFunction = _PyUnicode_AsString(poFrame->f_code->co_name);
-  int iLine=PyFrame_GetLineNumber(poFrame);
-  _ZOO_DEBUG(str,pccModule,pccFunction,iLine);
-  Py_RETURN_NONE;
+    char *str;
+
+    // Expecting a single string argument
+    if (!PyArg_ParseTuple(args, "s", &str)) {
+        fprintf(stderr, "Incorrect arguments to debug, only strings are supported\n");
+        Py_RETURN_NONE;
+    }
+
+    const char* pccModule = "unknown";
```

Comment:
The following variables are defined twice:
* `pccModule`
* `pccFunction`
* `iLine`